components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissions.kt

```diff
@@ -7,31 +7,64 @@ package mozilla.components.feature.sitepermissions
 import android.os.Parcel
 import android.os.Parcelable
 import mozilla.components.feature.sitepermissions.SitePermissions.Status.NO_DECISION
+import mozilla.components.feature.sitepermissions.SitePermissionsStorage.Permission
+import mozilla.components.feature.sitepermissions.SitePermissionsStorage.Permission.BLUETOOTH
+import mozilla.components.feature.sitepermissions.SitePermissionsStorage.Permission.CAMERA
+import mozilla.components.feature.sitepermissions.SitePermissionsStorage.Permission.MICROPHONE
+import mozilla.components.feature.sitepermissions.SitePermissionsStorage.Permission.NOTIFICATION
+import mozilla.components.feature.sitepermissions.SitePermissionsStorage.Permission.LOCATION
+import mozilla.components.feature.sitepermissions.SitePermissionsStorage.Permission.LOCAL_STORAGE
 import mozilla.components.feature.sitepermissions.db.StatusConverter
 
 /**
  * A site permissions and its state.
  */
 data class SitePermissions(
     val origin: String,
-    val location: Status = NO_DECISION,
-    val notification: Status = NO_DECISION,
-    val microphone: Status = NO_DECISION,
-    val camera: Status = NO_DECISION,
-    val bluetooth: Status = NO_DECISION,
-    val localStorage: Status = NO_DECISION,
+    val permissions: Map<Permission, Status> = emptyMap(),
     val savedAt: Long
 ) : Parcelable {
 
+    // TODO: Remove, and replace all usages
```

Comment:
Adding `@Deprecated` here is fine, so that you don't need to replace every single usage right now.

Reply:
I tried that, but it won't compile because warnings are treated as errors.

Reply:
It happens because we are using this constructor internally.

https://github.com/mozilla-mobile/android-components/blob/0e3b8ad582c974993d4babdeee2230065c07e2cc/components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/db/SitePermissionsEntity.kt#L59

Reply:
Ok I'll go ahead and deprecate this, then remove any internal usages. Just wanted to make sure I was headed in the right direction in general